Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageRepliesView.swift

```diff
@@ -116,3 +116,42 @@ public struct MessageRepliesView<Factory: ViewFactory>: View {
         }
     }
 }
+
+/// Lazy view that uses the message controller to fetch the parent message before creating message replies view.
+/// This is need when the parent message is not available in the local cache.
+/// Changing the `parentMessage` to `nil` in the `MessageRepliesView` would case multiple changes including breaking changes.
+struct LazyMessageRepliesView<Factory: ViewFactory>: View {
+    @ObservedObject private var parentMessageObserver: ChatMessageController.ObservableObject
+
+    var factory: Factory
+    var channel: ChatChannel
+    var message: ChatMessage
+
+    init(
+        factory: Factory,
+        channel: ChatChannel,
+        message: ChatMessage,
+        parentMessageController: ChatMessageController
+    ) {
+        parentMessageObserver = parentMessageController.observableObject
+        self.factory = factory
+        self.channel = channel
+        self.message = message
+        parentMessageObserver.controller.synchronize()
+    }
+
+    var body: some View {
+        Group {
+            if let parentMessage = parentMessageObserver.message {
+                factory.makeMessageRepliesShownInChannelView(
+                    channel: channel,
+                    message: message,
+                    parentMessage: parentMessage,
+                    replyCount: parentMessage.replyCount
+                )
+            } else {
+                EmptyView()
+            }
+        }
+    }
+}
```

Comment:
Did you also explore the way of fetching the missing message(s) in the `ChatChannelDataSource`? Thinking about fetching a page, then checking parentMessageIds and comparing, if the parent message id is part of the currently fetched list of message ids. If not, then fetching it before making the delegate callback. That should take away that delay when the thread reply appears. Could be an alternative, but not sure if better than what you have at the moment (might get messy).

Reply:
IMO the extra complexity is not worth it. Since this is meant to be temporary and although yes it has a bit of delay for the view to appear, this only happens for "Show in channel" messages which is not used very often either way 🤔

Reply:
I agree, not worth to complicate this further.